spec/http.md

```diff
@@ -0,0 +1,51 @@
+HTTP
+====
+
+## Resources
+
+Each EC2 instance MUST present the following (GET) resources:
+
+    [/service/healthcheck](#healthcheck)
+    [/service/gtg](#gtg)
+    [/service/dependencies](#dependencies)
+
+These resources should be presented on port 8080 (not port 80).
+
+### <a name="healthcheck">Healthcheck</a>
+
+Is the service running? Two possible responses:
+
+    200 "OK" (service is running)
+    50* (service is down - choose exact status code as appropriate)
+
+### <a name="gtg">GTG</a>
+
+    200 "OK" (service is able to usefully respond to requests)
+    50* (service is down - choose exact status code as appropriate)
+
+Is the service able to usefully respond to requests? (Typically, this
+indicates whether the service is running and its dependencies are
+available as expected.)
```

Comment:
Ah yes, we've got it the wrong way around! I do think there is value in having two different endpoints here, even if Amazon won't let you use a different check for ASG vs ELB membership (ideally, GTG - as defined by Beamly - would be used for ASG, and healthcheck for ELBs). Getting insight into dependencies from the apps own perspective is incredibly useful, even if we don't want to use it for automated checks.